client/my-sites/stats/stats-reditect-flow/index.tsx

```diff
@@ -19,47 +20,49 @@ const StatsRedirectFlow = () => {
 		getSiteOption( state, siteId, 'created_at' )
 	) as string;
 
-	const isCommercial = useSelector( ( state ) => getSiteOption( state, siteId, 'is_commercial' ) );
+	const { isFreeOwned, isPWYWOwned, isCommercialOwned } = useStatsPurchases( siteId );
 
 	const isSiteJetpackNotAtomic = useSelector( ( state ) =>
 		isJetpackSite( state, siteId, { treatAtomicAsJetpackSite: false } )
 	);
 
-	const { isFetching, data: purchaseRedirect } = useNoticeVisibilityQuery(
+	const { isFetching, data: purchaseNotPosponed } = useNoticeVisibilityQuery(
 		siteId,
 		'focus_jetpack_purchase'
 	);
 
+	const hasPlan = isFreeOwned || isPWYWOwned || isCommercialOwned;
 	// TODO: update the date to the release date when the feature is ready.
+	const qualifiedUser =
+		siteCreatedTimeStamp && new Date( siteCreatedTimeStamp ) > new Date( '2024-01-15' );
+
+	// to redirect the user can't have a plan purached and can't have the flag true, if either is true the user either has a plan or is postponing
 	const redirectToPurchase =
 		config.isEnabled( 'stats/checkout-flows-v2' ) &&
 		isSiteJetpackNotAtomic &&
-		purchaseRedirect &&
-		siteCreatedTimeStamp &&
-		new Date( siteCreatedTimeStamp ) > new Date( '2024-01-15' );
+		! hasPlan &&
+		purchaseNotPosponed &&
+		qualifiedUser;
 
 	const isRequesting = useSelector( ( state: object ) => isStatsNoticeSettingsFetching( state ) );
 	const dispatch = useDispatch();
 
-	// Only runs on mount.
 	useEffect( () => {
 		if ( isFetching ) {
 			// when react-query is fetching data
 			dispatch( requestStatNoticeSettings( siteId ) );
 		} else {
 			dispatch(
 				receiveStatNoticeSettings( siteId, {
-					focus_jetpack_purchase: purchaseRedirect,
+					focus_jetpack_purchase: purchaseNotPosponed,
 				} )
 			);
 		}
-	}, [ dispatch, redirectToPurchase, siteId, isFetching, purchaseRedirect ] );
+	}, [ dispatch, redirectToPurchase, siteId, isFetching, purchaseNotPosponed ] );
 
 	// render purchase flow for Jetpack sites created after February 2024
-	if ( ! isRequesting && redirectToPurchase && siteSlug ) {
-		page.redirect(
-			`/stats/purchase/${ siteSlug }?productType=${ isCommercial ? 'commercial' : 'personal' }`
-		);
+	if ( ! isFetching && ! isRequesting && redirectToPurchase && siteSlug ) {
```

Comment:
we have `isFetching` and `isRequesting` for the same state which is a little strange, but I don't think I have any suggestions 🤔